runtime/jvmti/jvmtiStartup.c

```diff
@@ -1061,6 +1066,169 @@ createXrunLibraries(J9JavaVM * vm)
+			/* compare with libName added the prefix/suffix */
+			if ((0 == strcmp(tmpLibName, libNameSo))
+				|| (0 == strcmp(tmpLibName, libNameAr))
+			) {
+				/* check if this object file path is part of libpath */
+				UDATA dirLen = tmp - fileName;
+				result = isFileNameInLibPath(fileName, dirLen, libpath);
+			}
+		} else {
+			/* check if libName contains a path separator */
+			if (isLibraryPath) {
+				/* compare the library directly with the object file loaded */
+				if (0 == strcmp(fileName, libName)) {
+					Trc_JVMTI_searchObjectFileFromLDInfo_library_object_file_found(fileName);
+					result = TRUE;
+				} else {
+					Trc_JVMTI_searchObjectFileFromLDInfo_library_object_file_not_found(fileName, libName);
+				}
+			} else {
+				/* libName has no path separator, compare the whole file names */
+				if (0 == strcmp(tmpLibName, libName)) {
+					/* check if this object file path is part of libpath */
+					UDATA dirLen = tmp - fileName;
+					result = isFileNameInLibPath(fileName, dirLen, libpath);
+				}
+			}
+		}
+	}
+	return result;
+}
+
+/**
+ * Retrieve a list of all object files loaded for the current process via loadquery(L_GETINFO).
+ * This helper method reallocates the buffer if the initial allocation is insufficient.
+ * The caller is responsible for freeing the buffer allocated.
+ *
+ * @param[in] vm Java VM
+ *
+ * @return a pointer to struct ld_info holding the object files if succeeded, otherwise NULL
+ */
+static struct ld_info * loadqueryGetObjectFiles(J9JavaVM *vm)
+{
+	PORT_ACCESS_FROM_JAVAVM(vm);
+	/* Use the same initial buffer size as omrintrospect_backtrace_symbols_raw()
+	 * in omr/port/aix/omrosbacktrace_impl.c.
+	 */
+	UDATA objectFileSize = 150 * sizeof(struct ld_info);
+	BOOLEAN success = FALSE;
+	void *tmpBuffer = j9mem_allocate_memory(objectFileSize, J9MEM_CATEGORY_JVMTI);
+	if (NULL != tmpBuffer) {
+		Trc_JVMTI_loadqueryGetObjectFiles_allocate_memory(tmpBuffer, objectFileSize);
+		for (;;) {
+			int rc = loadquery(L_GETINFO, tmpBuffer, objectFileSize);
+			if (-1 != rc) {
+				/* loadquery() completed successfully */
+				break;
+			}
+			if (ENOMEM == errno) {
+				void *newBuffer = NULL;
+				/* insufficient buffer size, increase it */
+				objectFileSize *= 2;
+				newBuffer = j9mem_reallocate_memory(tmpBuffer, objectFileSize, J9MEM_CATEGORY_JVMTI);
+				if (NULL == newBuffer) {
+					Trc_JVMTI_loadqueryGetObjectFiles_OOM(objectFileSize);
+					j9mem_free_memory(tmpBuffer);
+					tmpBuffer = NULL;
+					break;
+				}
+				tmpBuffer = newBuffer;
+				Trc_JVMTI_loadqueryGetObjectFiles_allocate_memory(tmpBuffer, objectFileSize);
+			} else {
+				/* other errors - EINVAL or EFAULT */
+				Trc_JVMTI_loadqueryGetObjectFiles_loadquery_errno(errno);
+				j9mem_free_memory(tmpBuffer);
+				tmpBuffer = NULL;
+				break;
+			}
+		}
+	}
+
+	if (NULL == tmpBuffer) {
+		Trc_JVMTI_loadqueryGetObjectFiles_OOM(objectFileSize);
+		vm->internalVMFunctions->setNativeOutOfMemoryError(vm->mainThread, 0, 0);
```

Comment:
Suggest the call to `setNativeOutOfMemoryError()` belongs in the caller.

Reply:
There is another failing case that is not OOM when `loadquery()` returns an `errno` other than `ENOMEM`.
Moved `setNativeOutOfMemoryError()` to individual OOM cases.